DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/FloorConversionToSpeckle.cs

```diff
@@ -9,7 +15,94 @@ namespace Speckle.Converters.Common;
 // clould be anywhere or all in once place - rooting through box 274 for something you need, when said box has a miriad different
 // and unrelated items, is no fun. Plus when you need that item, you end up bringing out the whole box/
 [NameAndRankValue(nameof(DB.Floor), 0)]
-public class FloorConversionToSpeckle : IHostObjectToSpeckleConversion
+public class FloorConversionToSpeckle : BaseConversionToSpeckle<DB.Floor, SOBR.RevitFloor>
 {
-  public Base Convert(object target) => throw new SpeckleConversionException();
+  private readonly IRawConversion<DB.CurveArrArray, List<SOG.Polycurve>> _curveArrArrayConverter;
+  private readonly IRawConversion<DB.Level, SOBR.RevitLevel> _levelConverter;
+  private readonly ParameterValueExtractor _parameterValueExtractor;
+  private readonly ParameterObjectAssigner _parameterObjectAssigner;
+  private readonly DisplayValueExtractor _displayValueExtractor;
+  private readonly SlopeArrowExtractor _slopeArrowExtractor;
+
+  public FloorConversionToSpeckle(
+    IRawConversion<CurveArrArray, List<Polycurve>> curveArrArrayConverter,
+    IRawConversion<Level, RevitLevel> levelConverter,
+    ParameterValueExtractor parameterValueExtractor,
+    ParameterObjectAssigner parameterObjectAssigner,
+    DisplayValueExtractor displayValueExtractor,
+    SlopeArrowExtractor slopeArrowExtractor
+  )
+  {
+    _curveArrArrayConverter = curveArrArrayConverter;
+    _levelConverter = levelConverter;
+    _parameterValueExtractor = parameterValueExtractor;
+    _parameterObjectAssigner = parameterObjectAssigner;
+    _displayValueExtractor = displayValueExtractor;
+    _slopeArrowExtractor = slopeArrowExtractor;
+  }
+
+  public override RevitFloor RawConvert(Floor target)
+  {
+    RevitFloor speckleFloor = new();
+
+    var sketch = (Sketch)target.Document.GetElement(target.SketchId);
+    List<SOG.Polycurve> profiles = _curveArrArrayConverter.RawConvert(sketch.Profile);
+
+    var type = target.Document.GetElement(target.GetTypeId()) as ElementType;
+    speckleFloor.family = type?.FamilyName;
+    speckleFloor.type = type?.Name;
+
+    // POC: https://spockle.atlassian.net/browse/CNX-9396
```

Comment:
Might be good to just add a comment as to what the ticket relates to, just a 1 liner for context